src/gleam/javascript/promise.gleam

```diff
@@ -5,6 +5,9 @@ import gleam/javascript/array.{Array}
 // TODO: labels
 pub type Promise(value)
 
+@external(javascript, "../../ffi.mjs", "newPromise")
+pub fn new(a: fn(fn(value) -> Nil) -> Nil) -> Promise(value)
```

Comment:
Could we have some tests for this please? We'll need to include a test to make sure that nested promises don't collapse.

Reply:
done